pkg/runtime/serializer/json/json.go

```diff
@@ -194,30 +194,59 @@ func (s *Serializer) RecognizesData(peek io.Reader) (bool, error) {
 	return ok, nil
 }
 
+// EncodesAsText returns true because both JSON and YAML are considered textual representations
+// of data. This is used to determine whether the serialized object should be transmitted over
+// a WebSocket Text or Binary frame. This must remain true for legacy compatibility with v1.1
+// watch over websocket implementations.
+func (s *Serializer) EncodesAsText() bool {
+	return true
+}
+
+// Framer is the default JSON framing behavior, with newlines delimiting individual objects.
+var Framer = jsonFramer{}
+
+type jsonFramer struct{}
+
 // NewFrameWriter implements stream framing for this serializer
-func (s *Serializer) NewFrameWriter(w io.Writer) io.Writer {
-	if s.yaml {
-		// TODO: needs document framing
-		return nil
-	}
+func (jsonFramer) NewFrameWriter(w io.Writer) io.Writer {
 	// we can write JSON objects directly to the writer, because they are self-framing
 	return w
 }
 
 // NewFrameReader implements stream framing for this serializer
-func (s *Serializer) NewFrameReader(r io.Reader) io.Reader {
-	if s.yaml {
-		// TODO: needs document framing
-		return nil
-	}
+func (jsonFramer) NewFrameReader(r io.Reader) io.Reader {
 	// we need to extract the JSON chunks of data to pass to Decode()
 	return framer.NewJSONFramedReader(r)
 }
 
-// EncodesAsText returns true because both JSON and YAML are considered textual representations
-// of data. This is used to determine whether the serialized object should be transmitted over
-// a WebSocket Text or Binary frame. This must remain true for legacy compatibility with v1.1
-// watch over websocket implementations.
-func (s *Serializer) EncodesAsText() bool {
-	return true
+// Framer is the default JSON framing behavior, with newlines delimiting individual objects.
+var YAMLFramer = yamlFramer{}
+
+type yamlFramer struct{}
+
+// NewFrameWriter implements stream framing for this serializer
+func (yamlFramer) NewFrameWriter(w io.Writer) io.Writer {
+	return &yamlFrameWriter{w: w}
+}
+
+// NewFrameReader implements stream framing for this serializer
+func (yamlFramer) NewFrameReader(r io.Reader) io.Reader {
+	// extract the YAML document chunks directly
+	return utilyaml.NewDocumentDecoder(r)
+}
+
+type yamlFrameWriter struct {
+	w    io.Writer
+	open bool
+}
+
+func (w *yamlFrameWriter) Write(data []byte) (n int, err error) {
+	if w.open {
+		if _, err := w.w.Write([]byte("\n---\n")); err != nil {
```

Comment:
Yes it is - this is multi-document separators.  Will link to formal doc and add godoc.